pvlib/ivtools/__init__.py

```diff
@@ -1,559 +1,7 @@
-    Gp = beta1 / (1.0 - Rs * beta1)
-    Rsh = 1.0 / Gp
-    IL = (1 + Gp * Rs) * beta0
-    # calculate I0
-    I0_vmp = _calc_I0(IL, i_mp, v_mp, Gp, Rs, nNsVth)
-    I0_voc = _calc_I0(IL, 0, v_oc, Gp, Rs, nNsVth)
-    if any(np.isnan([I0_vmp, I0_voc])) or ((I0_vmp <= 0) and (I0_voc <= 0)):
-        raise RuntimeError("Parameter extraction failed: I0 is undetermined.")
-    elif (I0_vmp > 0) and (I0_voc > 0):
-        I0 = 0.5 * (I0_vmp + I0_voc)
-    elif (I0_vmp > 0):
-        I0 = I0_vmp
-    else:  # I0_voc > 0
-        I0 = I0_voc
-    return (IL, I0, Rsh, Rs, nNsVth)
-
-
-def _system_of_equations_desoto(params, specs):
-    """Evaluates the systems of equations used to solve for the single
-    diode equation parameters. Function designed to be used by
-    scipy.optimize.root() in fit_sdm_desoto().
-
-    Parameters
-    ----------
-    params: ndarray
-        Array with parameters of the De Soto single diode model. Must be
-        given in the following order: IL, Io, a, Rsh, Rs
-    specs: tuple
-        Specifications of pv module given by manufacturer. Must be given
-        in the following order: Isc, Voc, Imp, Vmp, beta_oc, alpha_sc
-
-    Returns
-    -------
-    system of equations to solve with scipy.optimize.root().
-
-
-    References
-    ----------
-    [1] W. De Soto et al., "Improvement and validation of a model for
-    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88,
-    2006.
-
-    [2] John A Dufﬁe, William A Beckman, "Solar Engineering of Thermal
-    Processes", Wiley, 2013
-    """
-
-    # six input known variables
-    Isc, Voc, Imp, Vmp, beta_oc, alpha_sc, EgRef, dEgdT, Tref, k = specs
-
-    # five parameters vector to find
-    IL, Io, a, Rsh, Rs = params
-
-    # five equation vector
-    y = [0, 0, 0, 0, 0]
-
-    # 1st equation - short-circuit - eq(3) in [1]
-    y[0] = Isc - IL + Io * np.expm1(Isc * Rs / a) + Isc * Rs / Rsh
-
-    # 2nd equation - open-circuit Tref - eq(4) in [1]
-    y[1] = -IL + Io * np.expm1(Voc / a) + Voc / Rsh
-
-    # 3rd equation - Imp & Vmp - eq(5) in [1]
-    y[2] = Imp - IL + Io * np.expm1((Vmp + Imp * Rs) / a) \
-        + (Vmp + Imp * Rs) / Rsh
-
-    # 4th equation - Pmp derivated=0 - eq23.2.6 in [2]
-    # caution: eq(6) in [1] has a sign error
-    y[3] = Imp \
-        - Vmp * ((Io / a) * np.exp((Vmp + Imp * Rs) / a) + 1.0 / Rsh) \
-        / (1.0 + (Io * Rs / a) * np.exp((Vmp + Imp * Rs) / a) + Rs / Rsh)
-
-    # 5th equation - open-circuit T2 - eq (4) at temperature T2 in [1]
-    T2 = Tref + 2
-    Voc2 = (T2 - Tref) * beta_oc + Voc  # eq (7) in [1]
-    a2 = a * T2 / Tref  # eq (8) in [1]
-    IL2 = IL + alpha_sc * (T2 - Tref)  # eq (11) in [1]
-    Eg2 = EgRef * (1 + dEgdT * (T2 - Tref))  # eq (10) in [1]
-    Io2 = Io * (T2 / Tref)**3 * np.exp(1 / k * (EgRef/Tref - Eg2/T2))  # eq (9)
-    y[4] = -IL2 + Io2 * np.expm1(Voc2 / a2) + Voc2 / Rsh  # eq (4) at T2
-
-    return y
+from ivtools import fit_sde, fit_sdm, utility  # noqa: F401
```

Comment:
this needs to be either:

    from pvlib.ivtools import fit_sde, fit_sdm, utility  # noqa: F401

or

    from ..ivtools import fit_sde, fit_sdm, utility  # noqa: F401

Also need to fix `utility` improt in `fit_sde`